### ISSUE TITLE
metricstransform Regexp Config Causes Startup Failure in v0.104.0

### ISSUE THREAD

**Issue by fyuan1316** [open-telemetry/opentelemetry-collector#10585]
**Describe the bug**

Hi team,

After upgrading otel-collector-contrib from v0.100.0 to v0.104.0, we encountered a startup failure related to the regexp configuration in our component.

Below is our configuration  that worked in v0.100.0:

```yaml
processors:
  metricstransform:
    transforms:
      - include: http\.(.*)
        match_type: regexp
        action: update
        new_name: custom.http.$${1}
```

Errors Encountered

1. Using custom.http.$${1}:

```
Error: failed to resolve config: cannot resolve the configuration: environment variable "1" has invalid name: must match regex ^[a-zA-Z_][a-zA-Z0-9_]*$
2024/07/10 17:59:10 collector server run finished with error: failed to resolve config: cannot resolve the configuration: environment variable "1" has invalid name: must match regex ^[a-zA-Z_][a-zA-Z0-9_]*$
```

2. Changing to custom.http.${1}:

```
Error: failed to resolve config: cannot resolve the configuration: environment variable "1" has invalid name: must match regex ^[a-zA-Z_][a-zA-Z0-9_]*$
2024/07/10 18:00:31 collector server run finished with error: failed to resolve config: cannot resolve the configuration: environment variable "1" has invalid name: must match regex ^[a-zA-Z_][a-zA-Z0-9_]*$

```

3. Changing to custom.http.$1:

```
Error: failed to resolve config: cannot resolve the configuration: cannot convert the confmap.Conf: variable substitution using $VAR has been deprecated in favor of ${VAR} and ${env:VAR} - please update $1 or temporarily disable the confmap.unifyEnvVarExpansion feature gate
2024/07/10 18:01:19 collector server run finished with error: failed to resolve config: cannot resolve the configuration: cannot convert the confmap.Conf: variable substitution using $VAR has been deprecated in favor of ${VAR} and ${env:VAR} - please update $1 or temporarily disable the confmap.unifyEnvVarExpansion feature gate

```

Context

It seems there is a breaking change in the core regarding confmap environment variable expansion. Given the deprecation of variable substitution using $VAR, could you please advise on how to properly use regexp expressions in component configurations without disabling the confmap.unifyEnvVarExpansion feature gate?

Thank you for your assistance! :)

**Steps to reproduce**

**What did you expect to see?**

**What did you see instead?**

**What version did you use?**

0.104.0
**What config did you use?**

**Environment**

**Additional context**

I observed a similar issue in #10356 , but we are still encountering it in version 0.104.0.

**Comment by fyuan1316** [comment]
@TylerHelmuth @mx-psi do you have any suggestions? :)

**Comment by mx-psi** [comment]
This will be fixed in v0.105.0, until then you can disable the `confmap.unifyEnvVarExpansion` feature gate (by passing the `--feature-gates=-confmap.unifyEnvVarExpansion` CLI option).
